linuxkm/module_hooks.c

```diff
@@ -198,6 +198,58 @@ WC_MAYBE_UNUSED static int linuxkm_lkcapi_sysfs_deinstall_node(struct kobj_attri
     #include "linuxkm/lkcapi_glue.c"
 #endif
 
+int wc_linuxkm_check_for_intr_signals(void) {
+    static const int intr_signals[] = WC_LINUXKM_INTR_SIGNALS;
+    if (preempt_count() != 0)
+        return 0;
+
+#if defined(HAVE_FIPS) && defined(LINUXKM_LKCAPI_REGISTER)
+    /* ignore signals during FIPS startup sequence -- failed alg tests cause
+     * kernel panics on FIPS kernels.
+     */
+    if (linuxkm_lkcapi_registering_now)
```

Comment:
I think this is the only place `linuxkm_lkcapi_registering_now` is used, and it's guarded with HAVE_FIPS.

Should `linuxkm_lkcapi_registering_now` be wrapped with HAVE_FIPS everywhere?

Reply:
I don't see the need -- non-FIPS code might make use of that info in the future.  If anything, we might export it to the wider module, though for now that's not needed since the glue logic is basically all included into one giant file (`module_hooks.c`).